components/verification/ActionButton.tsx

```diff
@@ -0,0 +1,224 @@
+import { ActionButton } from '@/common/buttons'
+import { usePermission, useServerConfig } from '@/shell/ConfigurationContext'
+import {
+  AppBskyActorDefs,
+  ToolsOzoneVerificationGrantVerifications,
+} from '@atproto/api'
+import {
+  Popover,
+  PopoverButton,
+  PopoverPanel,
+  Transition,
+} from '@headlessui/react'
+import { CheckCircleIcon } from '@heroicons/react/24/solid'
+import { Fragment } from 'react'
+import { useVerifier } from './useVerifier'
+import { WorkspaceListData } from '@/workspace/useWorkspaceListData'
+import { isValidProfileViewDetailed } from '@/repositories/helpers'
+import { classNames } from '@/lib/util'
+
+export const BulkVerificationActionButton = ({
+  subjects,
+}: {
+  subjects?: WorkspaceListData
+}) => {
+  const { verifierDid } = useServerConfig()
+  const grantVerifications: ToolsOzoneVerificationGrantVerifications.VerificationInput[] =
+    []
+  const revokeUris: string[] = []
+
+  Object.values(subjects || {}).map((sub) => {
+    if (isValidProfileViewDetailed(sub.profile)) {
+      const verification = sub.profile.verification?.verifications?.find(
+        (v) => v.issuer == verifierDid,
+      )
+      if (verification) {
+        revokeUris.push(verification.uri)
+      } else {
+        grantVerifications.push({
+          subject: sub.profile.did,
+          handle: sub.profile.handle,
+          displayName: sub.profile.displayName || '',
+        })
```

Comment:
Does the backend reject/fail the verification if one already exists?  It's pretty easy to publish two verifications for the same user by mistake.  So having a backstop on ozone backend to help prevent that would be nice if it doesn't already exist.

Reply:
no we don't check existing verifications before creating new ones in the backend. frontend tries to make sure we don't show `grant verification` ui if one already exists but yeah, i can add the check on the backend too.

Reply:
Yeah, it's just that the UI is more likely to be out of sync with the actual state.  And nothing stops you from publishing multiple verification records for the same user.